waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -86,12 +102,15 @@ method atomicBatch*(g: OnchainGroupManager,
     var membersSeq = newSeq[Membership]()
     for i in 0 ..< idCommitments.len():
       var index = start + MembershipIndex(i)
-      debug "registering member", idCommitment = idCommitments[i], index = index
+      trace "registering member", idCommitment = idCommitments[i], index = index
       let member = Membership(idCommitment: idCommitments[i], index: index)
       membersSeq.add(member)
     await g.registerCb.get()(membersSeq)
 
   g.validRootBuffer = g.slideRootQueue()
+  let setMetadataRes = g.setMetadata()
+  if setMetadataRes.isErr():
+    error "failed to persist rln metadata", error=setMetadataRes.error
```

Comment:
I see that this method can either raise an exception or like in this case, just print the error.
For the future I think it worth reviewing this file and make all funcs/methods/procs to return `Result[T]` rather than raising exceptions.

Reply:
yes, you're right! for all the functions that return Future's, I abstain from returning results as 2 errors need to be handled then - will create a follow up pr for this

Reply:
but, setting the metadata is not a fatal error, we should be okay if we can't set it for any reason